### ISSUE TITLE
Can't Compile with xcode 5.1

### ISSUE THREAD

**Issue by jigfox** [eventmachine/eventmachine#505]
I found a lot information on rubygems and building native extensions with xcode 5.1

for most of the gems like json it worked like this described on [stackoverflow](http://stackoverflow.com/questions/22352838/ruby-gem-install-json-fails-on-mavericks-and-xcode-5-1-unknown-argument-mul)

But when I try this:

```
ARCHFLAGS=-Wno-error=unused-command-line-argument-hard-error-in-future gem install eventmachine
```

I get the following output:

```
/opt/boxen/rbenv/versions/2.1.1/bin/ruby extconf.rb
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/opt/boxen/rbenv/versions/2.1.1/bin/ruby
	--with-openssl-config
	--without-openssl-config
	--with-pkg-config
	--without-pkg-config
/opt/boxen/rbenv/versions/2.1.1/lib/ruby/2.1.0/mkmf.rb:456:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /opt/boxen/rbenv/versions/2.1.1/lib/ruby/2.1.0/mkmf.rb:541:in `try_link0'
	from /opt/boxen/rbenv/versions/2.1.1/lib/ruby/2.1.0/mkmf.rb:556:in `try_link'
	from /opt/boxen/rbenv/versions/2.1.1/lib/ruby/2.1.0/mkmf.rb:642:in `block in try_ldflags'
	from /opt/boxen/rbenv/versions/2.1.1/lib/ruby/2.1.0/mkmf.rb:635:in `with_ldflags'
	from /opt/boxen/rbenv/versions/2.1.1/lib/ruby/2.1.0/mkmf.rb:641:in `try_ldflags'
	from /opt/boxen/rbenv/versions/2.1.1/lib/ruby/2.1.0/mkmf.rb:1762:in `pkg_config'
	from extconf.rb:61:in `<main>'

extconf failed, exit code 1
```

It doesn't make sense because other gems compile.

Info about my system:

```
# OS X 10.9.2 with Xcode 5.1
rbenv -v #=> rbenv 0.4.0
ruby  -v #=> ruby 2.1.1p76 (2014-02-24 revision 45161) [x86_64-darwin13.0]
gcc -v
# Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-# dir=/usr/include/c++/4.2.1
# Apple LLVM version 5.1 (clang-503.0.38) (based on LLVM 3.4svn)
# Target: x86_64-apple-darwin13.1.0
# Thread model: posix

**Comment by sodabrew** [comment]
Is there any code or documentation change that needs to happen here?
The issue looks to be resolved / at least with user workarounds - May I close the issue?

**Comment by tmm1** [comment]
This is fixed in the 1.0.6 gem release.
